skimage/morphology/_max_tree.pyx

```diff
@@ -0,0 +1,459 @@
+"""_max_tree.pyx - building a max-tree from an image.
+
+This is an implementation of the max-tree, which is a morphological
+representation of the image. Many morphological operators can be built
+from this representation, namely attribute openings and closings.
+
+This file also contains implementations of max-tree based filters and
+functions to characterize the tree components. 
+"""
+
+import numpy as np
+
+from libc.stdlib cimport free, malloc, realloc
+from scipy.constants.constants import point
+from cProfile import label
+
+cimport numpy as np
+cimport cython
+
+ctypedef np.float64_t DTYPE_FLOAT64_t
+ctypedef np.int32_t DTYPE_INT32_t
+ctypedef np.uint32_t DTYPE_UINT32_t
+ctypedef np.uint64_t DTYPE_UINT64_t
+ctypedef np.int64_t DTYPE_INT64_t
+ctypedef np.uint8_t DTYPE_BOOL_t
+ctypedef np.uint8_t DTYPE_UINT8_t
+ctypedef np.int8_t DTYPE_INT8_t
+
+ctypedef fused dtype_t:
+    np.uint8_t
+    np.uint16_t
+    np.uint32_t
+    np.uint64_t
+    np.int8_t
+    np.int16_t
+    np.int32_t
+    np.int64_t
+    np.float32_t
+    np.float64_t
+
+cdef inline DTYPE_UINT64_t uint64_max(DTYPE_UINT64_t a, DTYPE_UINT64_t b): return a if a >= b else b
+cdef inline DTYPE_UINT64_t uint64_min(DTYPE_UINT64_t a, DTYPE_UINT64_t b): return a if a <= b else b
```

Comment:
If you have the time to check this: I believe Cython should be able to do this optimisation natively, so if you just write `max(a, b)` in the code it should be equally fast. Let me know if I'm wrong!

Reply:
OK.